### ISSUE TITLE
[Survey]Should CC have his own soundfiletype?

### ISSUE THREAD

**Issue by Wilma456** [dan200/ComputerCraft#281]
Do you think, CC should have his own soundfiletype? That makes it a lot easer to share soundfiles with other people.  There could be a function like sound,playFile(). We have with the normal paint format, a filetype, witch is accepted by the most graphic programs. So if we had a standart filetype, user can just upload songs they made and other people can play it with her favourite player. I only want to know, if the Community want this. The technical details coming after that.

Yes; Thumb Up
No : Thumb Down

**Comment by SquidDev** [comment]
To quote from #237:

> There are several players for the [Note Block Studio](http://www.stuffbydavid.com/mcnbs) format, such as BombBloke's [Note](https://github.com/theoriginalbit/MoarPeripherals/wiki/Note) program. This is pretty much the defacto standard,

As it's an already established file format, it makes sense to stick with it.

**Comment by dan200** [comment]
agreed

**Comment by lupus590** [comment]
Like @SquidDev quoted, Note Block Studio effectibly is that format.

If you are looking for a way to make a pull request then you could add an APi with some utility functions to process the format.

**Comment by Wilma456** [comment]
I see, threes a [API for this format](https://github.com/CrazedProgrammer/wave), but it is licensee under MIT, so you will have to put the license at the begin of the code. Maybe @CrazedProgrammer can made a PR with this API renamed as sound API.

**Comment by CrazedProgrammer** [comment]
I do not think `wave` should be included in the CC codebase, because nearly no program (outside of audio players and maybe games) are going to use it. It just doesn't really fit. It works way better as a separate API.  
Just having a speaker peripheral with a method for playing sounds is enough for CC.

**Comment by Wilma456** [comment]
@CrazedProgrammer Can I use your API to write a better API for CC?

**Comment by CrazedProgrammer** [comment]
@Wilma456 It's licensed under MIT, so you can, as long as it conforms to the license of course.

**Comment by lupus590** [comment]
@Wilma456 [info on the MIT license](https://choosealicense.com/licenses/mit/)

Summary, modification and redistribution are allowed provided as the license is included (comment at top of file will do). You may also want to link back to the original API, but you don't have to.

**Comment by Wilma456** [comment]
I know, what the MIT license is, but I think @dan200 doesn't want that text in Computercraft.

**Comment by lupus590** [comment]
I don't think @dan200  has indicated either way on whether to have the license info for code in the files. I can understand not wanting long licenses though. Perhaps something like this?

```lua
-- File name: example.lua
-- Author: Lupus590
-- Date: 25/05/2017 (DD/MM/YYYY)
-- License: MIT is available at https://choosealicense.com/licenses/mit/

print("example for licensing stuff")
```

**Comment by Wilma456** [comment]
I'm waiting for the OK from @dan200 before starting

**Comment by CrazedProgrammer** [comment]
I don't think you can just link to a page that describes the MIT license.  
The license states: `The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.`

**Comment by lupus590** [comment]
@CrazedProgrammer  I believe it's condoned when the full license would be impractical.

**Comment by CrazedProgrammer** [comment]
@lupus590 I doubt it. The license itself does not state that. If you do find a good source for that statement, please share.

**Comment by lupus590** [comment]
I've sound [evidence against](https://www.gnu.org/licenses/gpl-faq.en.html#WhyMustIInclude) my statement so maybe I was wrong.

**Comment by CrazedProgrammer** [comment]
You most likely were. If you really care about not having a 20 line license at the top of a source file you can just put it in a separate file, for example `licenses/wave`, so that the full license is still included in the distribution. In fact, that is already done with the LuaJ license in this repository.

**Comment by Wilma456** [comment]
Still waiting for the answer of @dan200

**Comment by apemanzilla** [comment]
https://github.com/dan200/ComputerCraft/issues/281#issuecomment-303770757

**Comment by lupus590** [comment]
@apemanzilla  Wilma wants Dans opinion on how to license the wave API that CrazedProgrammer made

**Comment by Restioson** [comment]
@Wilma456 
> I see, threes a API for this format, but it is licensee under MIT
This is really not a problem, since MIT is one of the freest licenses out there.

**Comment by dan200** [comment]
no
